File: Backend/schemas/request/timestamp.py
from pydantic import BaseModel, Field
from enum import Enum


class TimestampAction(str, Enum):
    to_datetime = "to_datetime"
    to_timestamp = "to_timestamp"


class TimestampConvertRequest(BaseModel):
    timestamp: int = Field(..., ge=0, le=9999999999, description="时间戳")
    action: TimestampAction = Field(..., description="转换类型")

File: Backup/tools_project_rust/src/models/timestamp.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TimestampAction {
    ToDatetime,
    ToTimestamp,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TimestampConvertRequest {
    pub timestamp: i64,
    pub action: TimestampAction,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TimestampConvertResponse {
    pub result: String,
    pub timestamp: i64,
    pub action: String,
}
